core/src/main/scala/pipeflow/dsl/nodes/Group.scala

```diff
@@ -1,17 +1,25 @@
-package pipeflow.core
+package pipeflow.dsl.nodes
 
-case class Group(
+import pipeflow.dsl.requirements.Requirement
+
+object Group {
+  def apply(id: String): Group = new Group(id)
+}
+
+case class Group private (
   id: String,
   name: Option[String] = None,
   requires: Seq[Requirement] = Seq.empty,
   children: Seq[Node] = Seq.empty
 ) extends Node {
 
+  def name(name: String): Group = this.copy(name = Some(name))
```

Comment:
have u considered using optics? http://julien-truffaut.github.io/Monocle/

Reply:
I've read about lenses a bit, and how useful are, specially for cases were there are nested immutable objects. But how much do you think it would improve that simple statement ?